lisa/analysis/rta.py

```diff
@@ -15,187 +15,493 @@
-            return match.group(1)
-
-        task_log_map = {
-            find_task_name(logfile): logfile
-            for logfile in rta_logs
-        }
-        return cls(task_log_map)
-
-    @classmethod
-    def from_dir(cls, log_dir):
-        """
-        Build a :class:`PerfAnalysis` from a folder path
-
-        :param log_dir: Folder containing RTApp log files
-        :type log_dir: str
-        """
-        rta_logs = glob.glob(os.path.join(
-            log_dir, cls.RTA_LOG_PATTERN.format(task='*'),
-        ))
-        return cls.from_log_files(rta_logs)
-
-    @classmethod
-    def from_task_names(cls, task_names, log_dir):
-        """
-        Build a :class:`PerfAnalysis` from a list of task names
-
-        :param task_names: List of task names to look for
-        :type task_names: list(str)
-
-        :param log_dir: Folder containing RTApp log files
-        :type log_dir: str
-        """
-        def find_log_file(task_name, log_dir):
-            log_file = os.path.join(log_dir, cls.RTA_LOG_PATTERN.format(task_name))
-            if not os.path.isfile(log_file):
-                raise ValueError('No rt-app logfile found for task [{}]'.format(
-                    task_name
-                ))
-            return log_file
-
-        task_log_map = {
-            task_name: find_log_file(task_name, log_dir)
-            for task_name in tasks_names
-        }
-        return cls(task_log_map)
-
-    @staticmethod
-    def _parse_df(logfile):
-        df = pd.read_csv(logfile,
-                sep='\s+',
-                skiprows=1,
-                header=0,
-                usecols=[1,2,3,4,7,8,9,10],
-                names=[
-                    'Cycles', 'Run' ,'Period', 'Timestamp',
-                    'Slack', 'CRun', 'CPeriod', 'WKPLatency'
-                ])
-        # Normalize time to [s] with origin on the first event
-        start_time = df['Timestamp'][0]/1e6
-        df['Time'] = df['Timestamp']/1e6 - start_time
-        df.set_index(['Time'], inplace=True)
+        super().__init__(trace)
+
+    def _task_filtered(self, df, task=None):
+        if not task:
+            return df
+        if task not in self.rtapp_tasks:
+            raise ValueError("Task [{}] is not an rt-app task: {}"
+                             .format(task, self.rtapp_tasks))
+
+        return df_filter_task_ids(df, [TaskID(None, task)],
+                                  pid_col='__pid', comm_col='__comm')
+
+    @memoized
+    def _get_rtapp_tasks(self):
+        task_ids = set()
+        for evt in self.trace.available_events:
+            if not evt.startswith('rtapp_'):
+                continue
+            fnc_name = "df_{}".format(evt)
+            df_getter = getattr(self.trace.analysis.rta, fnc_name)
+            df = df_getter()
+            for pid,name in df[['__pid', '__comm']].drop_duplicates().values:
+                task_ids.add(TaskID(pid, name))
+        return list(task_ids)
```

Comment:
Might want to use `sorted()` rather than `list()` if that property drives anything user-facing (terminal output, or even plots, otherwise the colours used in the plot with signals from multiple tasks will randomly change from one call to another for example).